filebeat/harvester/log_file.go

```diff
@@ -156,8 +138,12 @@ func (r *LogFile) errorChecks(err error) error {
 }
 
 func (r *LogFile) wait() {
-	// Wait before trying to read file wr.ch reached EOF again
-	time.Sleep(r.backoff)
+	// Wait before trying to read file again. File reached EOF.
+	select {
+	case <-r.done:
+		return
+	case <-time.After(r.backoff):
+	}
```

Comment:
It does not matter, as the first thing after `wait()` returns is that it checks again if the channel was closed.